### PR TITLE
Watch Feature Updates

### PR DESCRIPTION
**Pull request by ps2**
- Update IOB and COB immediately after carb/bolus send
- Adjust complication push budget to spend evenly before midnight
- Adjustments for Series 4, including rounded action buttons, correct text margin insets, and using system scroll haptics
- Improve efficiency of the chart redraw

### CODE REVIEW

File: Loop/Managers/WatchDataManager.swift [hunk 1]
```diff
@@ -301,3 +306,57 @@ extension WatchDataManager: WCSessionDelegate {
...
+            "* outstandingFileTransfers: \(outstandingFileTransfers)",
+            "* outstandingUserInfoTransfers: \(outstandingUserInfoTransfers)",
+            "* receivedApplicationContext: \(receivedApplicationContext)",
+            "* remainingComplicationUserInfoTransfers: \(remainingComplicationUserInfoTransfers)",
+            "* complicationUserInfoTransferInterval: \(round(complicationUserInfoTransferInterval.minutes)) min",
+            "* watchDirectoryURL: \(watchDirectoryURL?.absoluteString ?? "nil")",
+        ].joined(separator: "\n")
+    }
+
+    fileprivate var complicationUserInfoTransferInterval: TimeInterval {
```
Comment (mpangburn) [inline Loop/Managers/WatchDataManager.swift]: As I've been digging into this over the last week or two, I've found that midnight isn't the right estimate for the budget reset time—@elnjensen and I have been recording 5AM budget resets, but I don't think our two experiences are sufficient evidence to draw make a general conclusion.
I'm continuing to look into this, but we'd benefit from more logging here.

Comment (ps2) [reply Loop/Managers/WatchDataManager.swift]: You have been recording additional resets at 5am (so a midnight *and* a 5am reset?) or no midnight reset, and just the 5am one?

Comment (elnjensen) [reply Loop/Managers/WatchDataManager.swift]: It's 5:02 AM every day, but not midnight, at least for the two of us who have been tracking it - see #816 .   What we've observed is a daily reset at 5 AM, and then an additional reset (which doesn't change the daily schedule) whenever the Watch is fully charged.

Comment (ps2) [reply Loop/Managers/WatchDataManager.swift]: Ok.  The midnight target isn't an estimate of when the counter will reset, but an estimate of when the user will be asleep.  Still open to better algorithms for divvying this resource up, though.

**Pull request merged**